### ISSUE TITLE
ParseException with unusual dates

### ISSUE THREAD

**Issue by psibre** [marytts/marytts#349]
A ParseException is thrown if a date with an unexpected format is encountered:
```
java.lang.Exception: Module Preprocess: Problem processing the data.
	at marytts.server.Request.processOneChunk(Request.java:550)
	at marytts.server.Request.processOrLookupOneChunk(Request.java:385)
	at marytts.server.Request.process(Request.java:327)
	at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:261)
	at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:91)
	at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:138)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
	at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.inputReady(AsyncNHttpServiceHandler.java:316)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler.inputReady(BufferingHttpServiceHandler.java:131)
	at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:173)
	at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.text.ParseException: Unparseable date: "23.01.2001"
	at com.ibm.icu.text.DateFormat.parse(DateFormat.java:667)
	at marytts.language.en.Preprocess.expandDate(Preprocess.java:583)
	at marytts.language.en.Preprocess.expand(Preprocess.java:262)
	at marytts.language.en.Preprocess.process(Preprocess.java:178)
	at marytts.server.Request.processOneChunk(Request.java:548)
	... 19 more
```
The expected behavior would be to at least process the rest of the document, and a possible solution might be to apply different heuristics to parse the date (using, e.g., alternate delimiters).

**Comment by seblemaguer** [comment]
As far as I understand it seems that icu DateFormat doesn't really care about the locale you are using and use the system one which causes the failure....

**Comment by seblemaguer** [comment]
I don't really understand the description of this issue. Is it just a problem of a wrong parsing but a correct date format or a wrong date format ?

**Comment by seblemaguer** [comment]
Solved in #421
